src/khoj/routers/api_chat.py

```diff
@@ -73,6 +73,48 @@
 from khoj.routers.email import send_query_feedback
 
 
+class FilterRequest(BaseModel):
+    filename: str
+    conversation_id: str
+
+
+@api_chat.get("/conversation/file-filters/{conversation_id}", response_class=Response)
+@requires(["authenticated"])
+def get_file_filter(request: Request, conversation_id: str) -> Response:
+    conversation = ConversationAdapters.get_conversation_by_user(
+        request.user.object, conversation_id=int(conversation_id)
+    )
+    file_filters = conversation.file_filters
+    return Response(content=json.dumps(file_filters), media_type="application/json", status_code=200)
+
+
+@api_chat.post("/conversation/file-filters", response_class=Response)
+@requires(["authenticated"])
+def add_file_filter(request: Request, filter: FilterRequest):
+    try:
+        conversation = ConversationAdapters.get_conversation_by_user(
+            request.user.object, conversation_id=int(filter.conversation_id)
+        )
+        conversation.file_filters.append(filter.filename)
```

Comment:
You should add a verification step here that the `filename` exists in the user's indexed files.